scripts/shared/lib/deploy_bundle

```diff
@@ -0,0 +1,249 @@
+  namespace: ${MARKETPLACE_NAMESPACE}
+spec:
+  sourceType: grpc
+  image: ${INDEX_IMG}
+  displayName: Submariner Catalog Source
+  publisher: Submariner.io (Test)
+  updateStrategy:
+    registryPoll:
+      interval: 5m
+EOF
+
+  # Wait for the catalogSource Readiness
+  if ! (timeout 5m bash -c "until [[ $(kubectl --context="${cluster}" get catalogsource -n ${MARKETPLACE_NAMESPACE} submariner-catalog-source -o jsonpath='{.status.connectionState.lastObservedState}') -eq 'READY' ]]; do sleep 10; done"); then
+      echo "[ERROR] CatalogSource is not ready."
+      exit 1
+  fi
+
+  # Debug
+  kubectl -n ${MARKETPLACE_NAMESPACE} get catalogsource --ignore-not-found
+  kubectl -n ${MARKETPLACE_NAMESPACE} get pods --ignore-not-found
+  kubectl -n ${MARKETPLACE_NAMESPACE} get packagemanifests --ignore-not-found | grep 'Submariner Catalog Source' || true
+
+
+  # Create the OperatorGroup
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: submariner-group
+  namespace: ${SUBM_NS}
+spec:
+  targetNamespaces:
+    - ${SUBM_NS}
+EOF
+
+  # Debug
+  kubectl get og -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Catalog source created"
+}
+
+function install_bundle() {
+  local installPlan
+
+  # Delete previous catalogSource and Subscription
+  kubectl delete sub/submariner-subscription -n "${SUBM_NS}" --wait --ignore-not-found
+
+  # Create the Subscription (Approval should be Manual not Automatic in order to pin the bundle version)
+  echo "[INFO](${cluster}) Install the bundle..."
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: submariner-subscription
+  namespace: ${SUBM_NS}
+spec:
+  channel: ${CHANNEL}
+  installPlanApproval: Manual
+  name: ${OPERATOR_NAME}
+  source: submariner-catalog-source
+  sourceNamespace: ${MARKETPLACE_NAMESPACE}
+  startingCSV: ${OPERATOR_NAME}.v${VERSION}
+EOF
+
+  # Manual Approve
+  echo "[INFO](${cluster}) Approve the installPlan..."
+  kubectl wait --for condition=InstallPlanPending --timeout=5m -n "${SUBM_NS}" subs/submariner-subscription || (echo "[ERROR](${cluster}) InstallPlan not found."; exit 1)
+  installPlan=$(kubectl get subscriptions.operators.coreos.com submariner-subscription -n "${SUBM_NS}" -o jsonpath='{.status.installPlanRef.name}')
+  if [ -n "${installPlan}" ]; then
+    kubectl patch installplan -n "${SUBM_NS}" "${installPlan}" -p '{"spec":{"approved":true}}' --type merge
+  fi
+
+  # Debug
+  kubectl get sub -n "${SUBM_NS}" --ignore-not-found
+  kubectl get installplan -n "${SUBM_NS}" --ignore-not-found
+  kubectl get csv -n "${SUBM_NS}" --ignore-not-found
+  kubectl get pods -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Bundle installed"
+}
+
+function setup_broker() {
+  local brokerClientSecret
+
+  if ! (timeout 5m bash -c "until kubectl --context=${cluster} get crds brokers.submariner.io > /dev/null 2>&1; do sleep 10; done"); then
```

Comment:
Can't you use `with_retries` for this retrying logic?

Reply:
`with_retries` does not support timeout does it?